docs/001_develop/03_client-capabilities/015_layout-management/layout_03_js_api.mdx

```diff
@@ -0,0 +1,90 @@
+id: client-layout-management-js
+keywords: [ layout, layout management, Drag and drop, Resizable panels, Tabbed , Responsive , Customizable, Splitting, Stacking, Popouts, Nested , State persistence, Flexible, Container, Item, Component, Content item, Stack, Row, Column, Header, Tabs, Lifecycle methods, API, Resize events, Drag , Close events, Javascript method, function, config, dynamic]
+tags:
+- layout
+- layout management
+- Drag and drop
+- Resizable panels
+- Tabbed
+- Responsive
+- Customizable
+- Splitting
+- Stacking
+- Popouts
+- Nested
+- State persistence
+- Flexible
+- Container
+- Item
+- Component
+- Content item
+- Stack
+- Row
+- Column
+- Header
+- Tabs
+- Lifecycle methods
+- API
+- Resize events
+- Drag
+- Close events
+- Imperative HTML
+- method
+- function
+- config
+- dynamic
+sidebar_position: 15
+---
+
+The JavaScript API is accessed through the methods on the root layout object and allows for saving/loading the layout state, and dynamically adding items to the layout at runtime.
+```typescript
+@customElement({
+  name: 'my-layout',
+  template: html<MyLayout>`<rapid-layout ${ref('layout')}></rapid-layout>`,
+})
+export class MyLayout extends GenesisElement {
+  layout: FoundationLayout; // notice the TypeScript type is `FoundationLayout`
+  doSomething() {
+    this.layout.addItem( ... ) // access the layout imperatively
+  }
+}
+```
+
+Before discussing the API it's useful to understand a number of concepts.
+
+## Dynamic registration and adding items
+
+To have a pane displayed on the layout system, it must be registered with the layout system. When using the [declarative API](./layout_02_html_api.mdx), the layout system takes care of this for you, but as you start to add items dynamically and then serialise the layout, you need to consider which panes are registered. See this contained example <!-- TODO link -->, which allows the user to add pre-determined items to the layout dynamically.
+
+:::tip
+If you are only using the declarative API, and not using any dynamic integrations with JavaScript, then you shouldn't need to set the registration names of any items, as all the same items will be registered when you load a previously saved layout. If you are dynamically adding items as well, it is highly recommended to set the registration names of items manually. This makes it easier to figure out what is and is not registered.
+* When using the declarative API, use the `registration` attribute on the `<foundation-layout-item>` component.
+* When using the JavaScript API, set the `registration` optional parameter on the registered element config <!-- TODO link -->.
+:::
+
+## Saving and loading layout
+
+The JavaScript API can be used to save and load layout states manually. This only describes the state of the dynamic layout itself. It is the responsibility of each component within the layout to serialise its own state, if required. To enable autosaving the layout <!-- TODO link -->.
+
+## Methods
```

Comment:
This section will be fully linked to API docs when they're added in a subsequent ticket